src/Interfaces/LazySet.jl

```diff
@@ -878,3 +880,106 @@ true
 ```
 """
 isconvextype(X::Type{<:LazySet}) = false
+
+"""
+    surface(X::LazySet{N}) where {N}
+
+Compute the surface area of a set.
+
+### Input
+
+- `X` -- set
+
+### Output
+
+A number representing the surface area of `X`.
+"""
+function surface(X::LazySet{N}) where {N}
+    if dim(X) == 2
+        return area(X)
+    else
+        throw(ArgumentError("the surface function is only implemented for " *
+                    "two-dimensional sets, but the given set is $(dim(X))-dimensional"))
+    end
+end
+
+"""
+    area(X::LazySet{N}) where {N}
+
+Compute the area of a two-dimensional polytopic set using the Shoelace formula.
+
+### Input
+
+- `X` -- two-dimensional set
+
+### Output
+
+A number representing the area of `X`.
+
+### Notes
+
+This algorithm is applicable to any lazy set `X` such that its list of vertices,
+`vertices_list`, can be computed.
+
+### Algorithm
+
+Let `m` be the number of vertices of `X`. The following instances are considered:
+
+- `m = 0, 1, 2`: the output is zero.
+- `m = 3`: the triangle case is computed using the Shoelace formula with 3 points.
+- `m = 4`: the quadrilateral case is obtained by the factored version of the Shoelace
+           formula with 4 points.
+
+Otherwise, the general Shoelace formula is used; for detals see the wikipedia
+article [Shoelace formula](https://en.wikipedia.org/wiki/Shoelace_formula).
+"""
+function area(X::LazySet{N}) where {N}
+    @assert dim(X) == 2 "this function only applies to two-dimensional sets, " *
+    "but the given set is $(dim(X))-dimensional"
+
+    vlist = vertices_list(X)
+    m = length(vlist)
+
+    if m <= 2
+        return zero(N)
+    end
+
+    if m == 3 # triangle
+        res = _area_triangle(vlist)
+
+    elseif m == 4 # quadrilateral
+        res = _area_quadrilateral(vlist)
+
+    else # general case
+        res = _area_polygon(vlist)
+    end
+
+    return res
+end
+
+function _area_triangle(v::Vector{VN}) where {N, VN<:AbstractVector{N}}
+    A = v[1]
+    B = v[2]
+    C = v[3]
+    res = A[1] * (B[2] - C[2]) + B[1] * (C[2] - A[2]) + C[1] * (A[2] - B[2])
+    return abs(res/2)
+end
+
+function _area_quadrilateral(v::Vector{VN}) where {N, VN<:AbstractVector{N}}
+    A = v[1]
+    B = v[2]
+    C = v[3]
+    D = v[4]
+    res = A[1] * (B[2] - D[2]) + B[1] * (C[2] - A[2]) + C[1] * (D[2] - B[2]) + D[1] * (A[2] - C[2])
+    return abs(res/2)
+end
+
+function _area_polygon(v::Vector{VN}) where {N, VN<:AbstractVector{N}}
+    m = length(v)
+    res = zero(N)
+    for i in 1:m-1
+        @inbounds res += v[i][1] * v[i+1][2] - v[i+1][1] * v[i][2]
+    end
+    @inbounds res += v[m][1] * v[1][2] - v[1][1] * v[m][2]
```

Comment:
```suggestion
    @inbounds res = v[m][1] * v[1][2] - v[1][1] * v[m][2]
    for i in 1:m-1
        @inbounds res += v[i][1] * v[i+1][2] - v[i+1][1] * v[i][2]
    end
```
Saves one line and one addition :muscle:

Reply:
I expect that the formula with only one multiplication per iteration is much more efficient. Is there a reason why you chose this implementation?

Reply:
:) i won't stop you to try and pull request a more efficient implementation, but for this PR the branch is fairly good i think.

> Is there a reason why you chose this implementation?

no.

Reply:
What about the code suggestion at the top?

Reply:
if you insist :)

Reply:
for the record i tried this version today and it is actually *slower*, 129ns vs 100ns for 100 for a random polygon with 100 vertices.

```julia
function _area_polygon2(v::Vector{VN}) where {N, VN<:AbstractVector{N}}
    m = length(v)
    @inbounds res = (v[1][1] + v[m][1]) * (v[1][2] - v[m][2])
    for i in 1:m-1
        @inbounds res += (v[i+1][1] + v[i][1]) * (v[i+1][2] - v[i][2])
    end
    return abs(res/2)
end
```